musicround/helpers/spotify_client_manager.py

```
@@ -0,0 +1,83 @@
+Updated core.py with improved Spotify OAuth handling
+"""
+import logging
+import datetime
+from flask import current_app, redirect, url_for, flash, session
+from flask_login import current_user
+from musicround.helpers.spotify_debug import DebugSpotifyClient
+from musicround.models import db
+
+logger = logging.getLogger("spotify.token")
+
+def ensure_valid_spotify_token(session, current_user):
```

Ruff (0.11.9)

Redefinition of unused `session` from line 6

(F811)